### PR TITLE
add services for get and set language

### PR DESCRIPTION
**Pull request by kochigami**
reference: http://doc.aldebaran.com/2-4/naoqi/audio/altexttospeech-api.html#altexttospeech-api
```ALTextToSpeechProxy::setLanguage``` and ```ALTextToSpeechProxy::getLanguage```

requires: https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/32

I tried it with pepper, naoqi 2.5.5.5, ros indigo 

```
roslaunch naoqi_driver naoqi_driver.launch
kochigami@kochigami-ThinkPad-T450:~/catkin_ws/src/naoqi_driver/src/services$ rosservice list | grep lang
/naoqi_driver/get_language
/naoqi_driver/set_language

kochigami@kochigami-ThinkPad-T450:~/catkin_ws/src/naoqi_driver/src/services$ rosservice call /naoqi_driver/set_language "language: 'English'" 
success: True

kochigami@kochigami-ThinkPad-T450:~/catkin_ws/src/naoqi_driver/src/services$ rosservice call /naoqi_driver/get_language "{}" 
language: English
```

### CODE REVIEW

File: src/helpers/driver_helpers.cpp [hunk 1]
```diff
@@ -194,6 +194,59 @@ const naoqi_bridge_msgs::RobotInfo& getRobotInfo( const qi::SessionPtr& session
...
 
+/** Function that sets language for robot
+ */
+static std_srvs::Empty& setLanguageLocal( const qi::SessionPtr& session, naoqi_bridge_msgs::SetLanguageRequest req)
+{
+  static qi::Url robot_url;
+
+  robot_url = session->url();
+
+  // Get the robot type
```
Comment (kochigami) [inline src/helpers/driver_helpers.cpp]: memo: need to modify this

File: src/helpers/driver_helpers.cpp [hunk 2]
```diff
@@ -194,6 +194,59 @@ const naoqi_bridge_msgs::RobotInfo& getRobotInfo( const qi::SessionPtr& session
...
+/** Function that gets language set at robot
+ */
+  static std::string& getLanguageLocal( const qi::SessionPtr& session)
+{
+  static qi::Url robot_url;
+  static std::string language;
+
+  robot_url = session->url();
+
+  // Get the robot type
```
Comment (kochigami) [inline src/helpers/driver_helpers.cpp]: memo: need to modify this

**Comment by nlyubova** [comment]
+1 fine for me. @suryaambrose what do you think?

**Comment by nlyubova** [comment]
@kochigami can you do a pull request with a data format to nao_interaction?

**Comment by kochigami** [comment]
- use srv files (https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/38) instead of https://github.com/ros-naoqi/naoqi_bridge_msgs/pull/32

@nlyubova 

> can you do a pull request with a data format to nao_interaction?

I'm sorry for my late response.
Could you tell me where I should refer to in `nao_interaction`?
I couldn't find the related function of setting/ getting a language.

**Comment by kochigami** [comment]
confirmed with:
- [x] Pepper + indigo
- [ ] Pepper + kinetic
- [x] NAO + indigo
- [ ] NAO + kinetic

**Comment by kochigami** [comment]
I found there are some errors when executing it with ros kinetic (both NAO and Pepper).

I will report and update tomorrow.
I'm sorry for lack of information.

**Comment by nlyubova** [comment]
Thank you so much for testing!
Please keep us updated.

**Comment by kochigami** [comment]
I'm trying to fix this pull-request, but it may take a long time...
I'm not good at C++. I'm sorry.
My working tree is here:
https://github.com/kochigami/naoqi_driver/pull/1

problem: `rosservice call /naoqi_driver/get_language` is not executed correctly. 

status:
1. when executing it on kinetic: (https://github.com/ros-naoqi/naoqi_driver/pull/87/commits/069536b84137a3b9a28f489eab10ec4b0f0ac789)
`rosservice call /naoqi_driver/get_language "{}"`
=> `ERROR: transport error completing service call: unable to receive data from sender, check sender's logs for details`

I found `resp.data = helpers::driver::getLanguage(session_, req);` in `get_language.cpp` was never executed and the error above appeared.

2. when executing it on kinetic (https://github.com/kochigami/naoqi_driver/pull/1/commits/68fb075341cb8ea7bd9be4d0878f66f38d99e03c)
`rosservice call /naoqi_driver/get_language "{}"` returns language name,
but I found `getLanguageLocal` function in `drivers_helpers.cpp` is executed only once.
```
rosservice call /naoqi_driver/set_language "data: "English"` => pepper speaks English
rosservice call /naoqi_driver/get_language "{}" => English

rosservice call /naoqi_driver/set_language "data: "Japanese"`=> pepper speaks Japanese
rosservice call /naoqi_driver/get_language "{}" => English, This is a problem.
```

**Comment by kochigami** [comment]
I modified my pull request as for the problem above.
(not using `const`and `set/getLanguageLocal` function)
If there is any problem, please let me know.

I confirmed with:
- [x] Pepper (NAOqi 2.5.5.5) + ros indigo 
- [x] Pepper (NAOqi 2.5.5.5) + ros kinetic 
- [x] NAO (NAOqi 2.1.4) + ros indigo
- [x] NAO (NAOqi 2.1.4) + ros kinetic

**Comment by nlyubova** [comment]
Thank you!
Just merged required ros-naoqi/naoqi_bridge_msgs#38

**Pull request merged**